mmv1/templates/terraform/constants/router_nat.go.erb

```diff
@@ -88,3 +88,12 @@ func computeRouterNatSubnetworkHash(v interface{}) int {
 	
 	return schema.HashString(NameFromSelfLinkStateFunc(name)) + sourceIpRangesHash + secondaryIpRangeHash
 }
+
+func computeRouterNatIPsHash(v interface{}) int {
+	val := (v.(string))
+	newParts := strings.Split(val, "/")
+	if len(newParts) == 1 {
+		return schema.HashString(newParts[0])
+	}
+	return schema.HashString(GetResourceNameFromSelfLink(val))
```

Comment:
I believe this leaves out the `project` part of the value.

So it could compare two self_links that include different projects and conclude that they are the same. 

For example:
"projects/my-project/addresses/address1"
And
"projects/another-project/addresses/address1"
would both produce the same hash.

Can you check to see if it is possible to set this field to reference a resource outside the resource's project? If that is possible we may need another solution

Reply:
Oh, the actual request is only to handle both ids and self_links. That should be doable here and we can not support the name field directly.

Does that make sense?

Reply:
I got it, does the doc have to specify that `name` attribute not supported or validatefunc needs to be added ?

Reply:
I don't think so

Reply:
It turned out cross project natIPs isn't be allowed at the API level, per discussion with @slevenick no validatefunc is needed to limit reference natIPs by `name`. Added TCs for both refer by id & name